### PR TITLE
sql: add list_agg

### PR DESCRIPTION
**Pull request by mjibson**

### CODE REVIEW

**Comment by benesch** [comment]
Deferring review to @sploiselle unless I'm needed! (Though looked like no surprises from a very quick skim.)

File: doc/user/data/sql_funcs.yml [hunk 1]
```diff
@@ -83,6 +83,9 @@
 - type: List
   description: List functions take [`list`](../types/list) arguments, and are [polymorphic](../types/list/#polymorphism).
   functions:
+  - signature: 'list_agg(x: any) -> L'
```
Comment (sploiselle) [inline doc/user/data/sql_funcs.yml:86]: nit: this indentation looks off from the line below it

**Review by sploiselle — APPROVED** [review]
lgtm modulo a couple of tests

File: test/sqllogictest/list.slt [hunk 2]
```diff
@@ -2311,3 +2311,86 @@ query T
 SELECT ('{{1}}'::int4_list list || '{2}'::int4 list)::text
 ----
 {{1},{2}}
+
+# list_agg
```
[the remark this answers is not in the document]
Comment (mjibson) [reply test/sqllogictest/list.slt:2315]: Added tests for these. Can't use an actual table because that'll go through symbiosis.

**Pull request merged**